dig_test.go

```diff
@@ -883,17 +883,14 @@ func TestCantProvideParameterObjects(t *testing.T) {
 	})
 
 	t.Run("pointer from constructor", func(t *testing.T) {
+		c := New()
 		type Args struct{ In }
 
 		args := &Args{}
 
-		c := New()
-		require.NoError(t, c.Provide(func() (*Args, error) {
-			return args, nil
-		}), "provide failed")
-		require.NoError(t, c.Invoke(func(a *Args) {
-			require.True(t, args == a, "args must match")
-		}), "invoke failed")
+		err := c.Provide(func() (*Args, error) { return args, nil })
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "can't provide pointers to parameter objects")
```

Comment:
There are plenty of tests that cover the pointers, like `*byte.Buffer`. This tries to cover the mis-use case when `In` is used instead of `Out`.

I just modified the previous test.